File: src/main/java/me/tbg/match/bot/MatchStartListener.java
package me.tbg.match.bot;

import java.awt.Color;
import java.time.Instant;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;

import net.dv8tion.jda.api.EmbedBuilder;
import tc.oc.pgm.api.map.MapInfo;
import tc.oc.pgm.api.match.Match;
import tc.oc.pgm.api.match.event.MatchStartEvent;

public class MatchStartListener implements Listener {

    private final DiscordBot bot;

    public MatchStartListener(DiscordBot bot) {
        this.bot = bot;
    }

    @EventHandler
    public void onMatchStart(MatchStartEvent event) {
        Match match = event.getMatch();
        MapInfo map = match.getMap();
        EmbedBuilder matchStartEmbed = createMatchStartEmbed(match, map);

        bot.setEmbedThumbnail(map, matchStartEmbed, bot);
        bot.sendMatchEmbed(matchStartEmbed, match);
        bot.storeMatchStartData(
                Long.parseLong(match.getId()),
                Instant.now().getEpochSecond(),
                match.getPlayers().size());
    }

    private EmbedBuilder createMatchStartEmbed(Match match, MapInfo map) {
        return new EmbedBuilder()
                .setColor(Color.WHITE.getRGB())
                .setTitle("Match #" + match.getId() + " has started!")
                .setDescription(bot.getMatchDescription(match))
                .addField("Map", map.getName(), true)
                .addField("Version", map.getVersion().toString(), true)
                .addField("Gamemodes", bot.getMapGamemodes(match).toUpperCase(), true)
                .addField("Created by", bot.getMapAuthors(match), true)
                .addField("Pools", bot.getMapPools(match), true)
                .addField("Objective", map.getDescription(), false)
                .addField("Participants", String.valueOf(match.getParticipants().size()), true)
                .addField("Observers", String.valueOf(match.getDefaultParty().getPlayers().size()), true);
    }
}

File: src/main/java/me/tbg/match/bot/MatchStartListener.kt
package me.tbg.match.bot

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.javacord.api.entity.message.embed.EmbedBuilder
import tc.oc.pgm.api.match.event.MatchStartEvent
import java.awt.Color
import java.time.Instant
import java.util.Locale

class MatchStartListener(private val bot: DiscordBot?) : Listener {
    @EventHandler
    fun onMatchStart(event: MatchStartEvent) {
        val match = event.match
        val map = match.map
        val matchStartEmbed = EmbedBuilder()
            .setColor(Color.WHITE)
            .setTitle("Match #" + match.id + " has started!")
            .setThumbnail(bot!!.getMapImageUrl(map))
            .setDescription(
                "Started at <t:"
                        + Instant.now().epochSecond
                        + ":f> with **"
                        + match.players.size
                        + (if (match.players.size == 1) " player" else " players")
                        + "** online."
            )
            .addInlineField("Map", map.name)
            .addInlineField("Version", map.version.toString())
            .addInlineField("Gamemodes", bot.getMapGamemodes(match).uppercase(Locale.getDefault()))
            .addInlineField("Created by", bot.getMapAuthors(match))
            .addInlineField("Pools", bot.getMapPools(match))
            .addField("Objective", map.description)
            .addInlineField("Participants", match.participants.size.toString())
            .addInlineField(
                "Observers", match.defaultParty.players.size.toString()
            )
            .addInlineField("Staff", bot.getOnlineStaffCount(match).toString())
            .setFooter("Map tags: " + map.tags.toString())
        bot.sendMatchEmbed(matchStartEmbed, event.match)
    }
}